AAP-35119 updated overview to tested topologies (#2511)
ansible/aap-docs: downstream/assemblies/topologies/assembly-overview-tested-deployment-models.adoc

= Overview of tested deployment models

> Red Hat tests {PlatformNameShort} {PlatformVers} with a defined set of topologies to give you opinionated deployment options. Red Hat recommends that customers always deploy all components of {PlatformNameShort} so that all features and capabilities are available for use without the need to take further action.

Reviewer: Suggestion based on RH supplementary guide regarding using "Red Hat suggests": 

```suggestion
Red Hat tests {PlatformNameShort} {PlatformVers} with a defined set of topologies to give you opinionated deployment options. Based on results from these tests, always deploy all components of {PlatformNameShort} so that all features and capabilities are available for use without the need to take further action.
```

> Red Hat tests the installation of {PlatformNameShort} {PlatformVers} based on a defined set of infrastructure topologies or reference architectures. Enterprise organizations should use one of the enterprise topologies for production deployments to ensure the highest level of uptime, performance, and continued scalability. Organizations or deployments that are resource constrained can use a "growth" topology.

Reviewer 2: Hey I'd recommend avoiding "should" here, maybe "can" or "must" instead?

IBM SG: "Whenever possible, rewrite in a more direct way. Do not use should to refer to an action that must be performed. "

Reviewer: @g-murray  - Yes, I considered adding that to the edits, but wording, "Whenever possible..." gave me pause. Tried the different options and felt like "should" was tolerable in this instance. Leaving it up to @emurtoug if she wants to make further changes after this PR is merged.

> Red Hat tests and recommends installing {PlatformNameShort} {PlatformVers} based on a defined set of infrastructure topologies or reference architectures. Enterprise organizations should use one of the enterprise topologies for production deployments to ensure the highest level of uptime, performance, and continued scalability. Organizations or deployments that are resource constrained can use a "growth" topology.

Reviewer: Same as above and based on RH supplementary guide: 

```suggestion
Install {PlatformNameShort} {PlatformVers} based on a defined set of infrastructure topologies or reference architectures. Enterprise organizations should use one of the enterprise topologies for production deployments to ensure the highest level of uptime, performance, and continued scalability. Organizations or deployments that are resource constrained can use a "growth" topology.
```

> It is possible to install the {PlatformNameShort} on different infrastructure topologies and with different environment configurations; Red Hat does not fully test topologies outside of published reference architectures, and recommends that all new deployments use a tested topology.

Reviewer: Suggestion to adhere to rules about "Red Hat recommends": 

```suggestion
It is possible to install the {PlatformNameShort} on different infrastructure topologies and with different environment configurations. Red Hat does not fully test topologies outside of published reference architectures. Use a tested topology for all new deployments.  
```
